server/kit/kitserver_test.go

```diff
@@ -15,10 +15,59 @@ import (
 	ocontext "golang.org/x/net/context"
 	"google.golang.org/grpc"
 
+	gserver "github.com/NYTimes/gizmo/server"
 	"github.com/NYTimes/gizmo/server/kit"
 	"github.com/NYTimes/gziphandler"
 )
 
+func TestKitServerHTTPMiddleware(t *testing.T) {
+	shutdownErrChan := make(chan error)
+	go func() {
+		// runs the HTTP _AND_ gRPC servers
+		shutdownErrChan <- kit.Run(&server{})
```

Comment:
Isn't `httptest.TestServer` the more normal Go way to bring up a test server?

https://golang.org/pkg/net/http/httptest/#NewServer

Reply:
Thats generally if you want to mock out a third party service's response. In this instance, we actually want to start up and hit our real server.

We may be able to just hit the server.ServeHTTP instead of actually listening on the port. Lemme see if I can simplify it.